src/init/features/ailogic/index.ts

```diff
@@ -0,0 +1,149 @@
+    choices: [
+      { name: "Android", value: "android" },
+      { name: "iOS", value: "ios" },
+      { name: "Web", value: "web" },
+    ],
+  });
+
+  // Ask for app namespace
+  let appNamespace: string;
+  if (platform === "android") {
+    appNamespace = await input({
+      message: "Enter your Android package name (e.g., com.example.myapp):",
+      validate: (input: string) => {
+        if (!input || !/^[a-zA-Z][a-zA-Z0-9_]*(\.[a-zA-Z][a-zA-Z0-9_]*)*$/.test(input)) {
+          return "Please enter a valid Android package name (e.g., com.example.myapp)";
+        }
+        return true;
+      },
+    });
+  } else if (platform === "ios") {
+    appNamespace = await input({
+      message: "Enter your iOS bundle ID (e.g., com.example.MyApp):",
+      validate: (input: string) => {
+        if (!input || !/^[a-zA-Z][a-zA-Z0-9_]*(\.[a-zA-Z][a-zA-Z0-9_]*)*$/.test(input)) {
+          return "Please enter a valid iOS bundle ID (e.g., com.example.MyApp)";
+        }
+        return true;
+      },
+    });
+  } else {
+    appNamespace = await input({
+      message: "Enter your web app name:",
+      validate: (input: string) => {
+        if (!input) {
+          return "Please enter a web app name";
+        }
+        return true;
+      },
+    });
+  }
+
+  // Ask about overwriting existing config files
+  const overwriteConfig = await confirm({
+    message: "Would you like to overwrite existing config files if they exist?",
+    default: false,
+  });
+
+  // Set up the feature info
+  if (!setup.featureInfo) {
+    setup.featureInfo = {};
+  }
+
+  setup.featureInfo.ailogic = {
+    appPlatform: platform,
+    appNamespace: appNamespace,
+    overwriteConfig: overwriteConfig,
+  };
+}
+
+/**
+ * AI Logic provisioning: auto-detects/creates project and app, provisions via API
+ */
+export async function actuate(setup: Setup, config: Config): Promise<void> {
+  const ailogicInfo = setup.featureInfo?.ailogic as AiLogicInfo;
+  if (!ailogicInfo) {
+    return;
+  }
+
+  try {
+    // 1. Determine app platform
+    const platform: SupportedPlatform =
+      ailogicInfo.appPlatform || (await detectAppPlatform(config.projectDir));
+
+    // 2. Check for config file conflicts
+    const configFilePath = getConfigFilePath(config.projectDir, platform);
+    if (fs.existsSync(configFilePath) && !ailogicInfo.overwriteConfig) {
+      throw new FirebaseError(
+        `Config file ${configFilePath} already exists. Use overwrite_config: true to update it.`,
+        { exit: 1 },
+      );
+    }
+
+    // 3. Build provisioning options
+    const provisionOptions = buildProvisionOptions(
+      setup.projectId,
+      platform,
+      ailogicInfo.appNamespace,
+    );
+
+    // 4. Provision Firebase app
+    const response = await provisionAiLogicApp(provisionOptions);
+
+    // 5. Extract project ID and update setup
+    const projectId = extractProjectIdFromAppResource(response.appResource);
```

Comment:
This step seems potentially very problematic. At this point, we may have used setup.projectId for other products.